src/View/Helper/Navigation/AbstractHelper.php

```diff
@@ -322,7 +324,7 @@ public function accept(PageInterface $page, bool $recursive = true): bool
                 [
                     'authorization' => $this->getUseAuthorization() ? $this->getAuthorization() : null,
```

Comment:
The $useAuthorization as well as the laminas-navigation $useAcl flags IMHO should have false default values.

Why?
If an authorization/acl instance has not been set (which should mutate the flag) then why attempt to consult the authorization component? If not set there is zero chance it can determine if the page should be accepted. In this case setting a default instance would also need to mutate the flag as well.

With that change for this implementation if neither have been set then 
```php
$this->getRenderInvisible()
```
would determine if the page is accepted and control would not need to be passed to the AcceptHelper.

Reply:
`$useAuthorization` IMHO offers the ability to disable the Authorization. This may be helpful if you have more than one navigation.

Reply:
If it is set to false as default, then you would enable it for the menus you wished to filter by Authorization. Which, seems the more intuitive use case, since it is supposed to work without Authorization as its standard mode of operation.

Consider
User A installs Mezzio navigation but has no need for Authorization on any nav components. With it set to true by default, then any logic behind the $useAuthorization check will run, even for their setup, since it is true by default. That should not be the default. If the component does not require Authorization, then the assumption is that $useAuthorization would be false and require mutation to enable it. Either explicitly via setUseAuthorization or implicitly via setAuthorization(...). 

User B
Installs Mezzio navigation and has three nav components. Two of which needs to be filtered by Authorization. Then those two have $useAuthorization set to true, and is filtered. The other nav component requires nothing additional since its the "standard" ie "default" use case.

History,
If you install a fresh build of Laminas MVC and install Laminas Navigation and configure a navigation component and have no authorization component installed or configured. For every entry in the container there will be a isAllowed event fired. Since there is no authorization component even present in the application it is not possible for its processing to alter execution flow, which means its superfluous processing.  

Conclusion,
Ultimately the decision is yours, just thought I should flesh out a little more of my thinking about it.